From 405263ab9642bb025b5af699de1b1599f7b0ffce Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Mon, 9 Dec 2013 16:19:12 +0100
Subject: [PATCH 034/117] sysprep: allow to specify globbing for --delete

Adapt the globbing part from the old --remote-path work previously
proposed for sysprep [1], allowing --delete to perform globbing when
deleting paths.

[1] https://www.redhat.com/archives/libguestfs/2013-October/msg00045.html

(cherry picked from commit ed4bcb119cb908e98ecc1107dcd8b740ee2f484f)
---
 sysprep/sysprep_operation_delete.ml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sysprep/sysprep_operation_delete.ml b/sysprep/sysprep_operation_delete.ml
index de78a87..59d5485 100644
--- a/sysprep/sysprep_operation_delete.ml
+++ b/sysprep/sysprep_operation_delete.ml
@@ -27,7 +27,9 @@ let add_paths path = paths := path :: !paths
 
 let path_perform g root =
   let paths = List.rev !paths in
-  List.iter g#rm_rf paths;
+  if paths <> [] then (
+    List.iter (fun glob -> Array.iter g#rm_rf (g#glob_expand glob)) paths
+  );
   []
 
 let op = {
-- 
2.5.0

